sqlmesh/core/config/loader.py

```diff
@@ -13,12 +13,15 @@
 from sqlmesh.utils.metaprogramming import import_python_file
 from sqlmesh.utils.yaml import load as yaml_load
 
+C = t.TypeVar("C", bound=Config)
+
 
 def load_configs(
-    config: t.Optional[str | Config],
+    config: t.Optional[str | t.Union[Config, C]],
+    config_type: t.Type[C],
```

Comment:
I couldn't decide if `config_type` should be a required field or not. I went with required for now, so that if more loader calls are added, the caller will be aware that they should pass in `config_type` for overriding to work on that load path. I could be convinced otherwise.

Reply:
I feel like having the default to be `Config` makes sense and simplifies the API